bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/QuantityType.java

```diff
@@ -74,7 +75,8 @@ public QuantityType(String value) {
         String[] constituents = value.split(UNIT_PATTERN);
 
         // getQuantity needs a space between numeric value and unit
-        String formatted = String.join(" ", constituents);
+        constituents = (String[]) ArrayUtils.add(constituents, constituents.length - 1, " ");
```

Comment:
If I understood it correctly, the problem was that the `UNIT_PATTERN` pattern matched twice, so what with the exponent we ended up having 3 constituents, right? How can we be sure then that `length - 1` is _always_ the right place to separate the numeric part from the units part?

Reply:
We assume that the unit is always the last part of the constructor String. The old implementation just did that by joining the two constituent by `" "`.
I'm not aware of other notation, but right now we would not support them.